wandb/sdk/launch/runner/abstract.py

```diff
@@ -142,6 +151,38 @@ def verify(self) -> bool:
             sys.exit(1)
         return True
 
+    def _fill_macros(self, project: LaunchProject, image: str) -> None:
```

Comment:
we should consider how this is implemented vs triggers. I dont want us to have 2 separate systems with separate semantics that will be confusing to the user

Reply:
Yeah they will have to be implemented separately but we should agree on some macro naming conventions @KyleGoyette